arbnode/delayed.go

```diff
@@ -105,9 +105,10 @@ func (b *DelayedBridge) GetAccumulator(ctx context.Context, sequenceNumber uint6
 }
 
 type DelayedInboxMessage struct {
-	BlockHash      common.Hash
-	BeforeInboxAcc common.Hash
-	Message        *arbostypes.L1IncomingMessage
+	BlockHash              common.Hash
+	BeforeInboxAcc         common.Hash
+	Message                *arbostypes.L1IncomingMessage
+	ParentChainBlockNumber uint64
```

Comment:
I think we'll need to make this
```
`rlp:"optional,omitempty"`
```
to keep it backwards compatible (I might be blanking on the syntax here). I'd also not fill it in if it's the same as BlockNumber to save a bit on storage, though that's optional. Either way, you should add a method to get the parent chain block number that checks this field but if nil falls back to BlockNumber, again for backwards compatibility with dbs without this field.